core/server/worker/src/test/java/alluxio/worker/block/MonoBlockStoreCommitBlockTest.java

```diff
@@ -0,0 +1,142 @@
+package alluxio.worker.block;
+
+import alluxio.exception.status.AlluxioStatusException;
+import alluxio.underfs.UfsManager;
+import alluxio.worker.block.io.BlockWriter;
+import alluxio.worker.block.meta.StorageDir;
+import io.grpc.Status;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+// This Test is a little different from the PagedBlockStoreCommitStore due to structure different.
+// MonoBlockStore.commitBlock() will call TieredBlockStore.commitBlocked() first as commitLocal, then will call BlockMasterClient.commitBlock() as commitMaster
+// TieredBlockStore.commitBlock() call TieredBLockStore.commitBlockInternal inside them wake the EventListener for listener.onCommitToLocal()
+// MonoBlockStore will wake the EventListener for listener.onCommitToMaster after BlockMasterClient.commitBlock() successes
+// In a nutshell two onCommit events weren't called in same domain
+public class MonoBlockStoreCommitBlockTest {
+    public MonoBlockStore mMonoBlockStore;
+    BlockMasterClientPool mockedBlockMasterClientPool;
+    BlockMasterClient mockedBlockMasterClient;
+    BlockMetadataManager mBlockMetadataManager;
+    BlockLockManager mBlockLockManager;
+    TieredBlockStore mTieredBlockStore;
+    private static final String FIRST_TIER_ALIAS = TieredBlockStoreTestUtils.TIER_ALIAS[0];
+    private StorageDir mTestDir1;
+    /** Rule to create a new temporary folder during each test. */
+    @Rule
+    public TemporaryFolder mTestFolder = new TemporaryFolder();
+
+    final Long sessionId = 1L;
+    final Long blockId = 2L;
+    int FIRST_TIER = 1;
+    // Maybe location should be asserted as well.
+    BlockStoreEventListener mListener;
+    @Before
+    public void setup() throws Exception {
+        File tempFolder = mTestFolder.newFolder();
+        TieredBlockStoreTestUtils.setupDefaultConf(tempFolder.getAbsolutePath());
+
+        mockedBlockMasterClientPool = mock(BlockMasterClientPool.class);
+        mockedBlockMasterClient = mock(BlockMasterClient.class);
+        when(mockedBlockMasterClientPool.acquire()).thenReturn(mockedBlockMasterClient);
+        doNothing().when(mockedBlockMasterClientPool).release(any());
+        mBlockLockManager = new BlockLockManager();
+        mBlockMetadataManager = BlockMetadataManager.createBlockMetadataManager();
+
+        mTestDir1 = mBlockMetadataManager.getTier(FIRST_TIER_ALIAS).getDir(0);
+
+        mListener = mock(BlockStoreEventListener.class);
+        doAnswer((i) -> {
+            assertEquals(2L, i.getArguments()[0]);
```

Comment:
Refer to the constant here
```suggestion
            assertEquals(BLOCK_ID, i.getArguments()[0]);
```